First pass at new hub and landing page design

BillWagner opened the pull request:
This uses the new hub page design and the new card layout.

[Internal review link](https://review.docs.microsoft.com/dotnet/index?branch=pr-en-us-15535)

BillWagner committed:
first draft

This uses the new hub page design and the new card layout.

BillWagner committed:
quote title with colon

BillWagner committed:
fix links

Youssef1313 commented on index.yml:
```diff
@@ -0,0 +1,235 @@
+conceptualContent:
+# itemType: architecture | concept | deploy | download | get-started | how-to-guide | learn | overview | quickstart | reference | tutorial | video | whats-new
+  title: ".NET: Free. Cross-platform. Open source" # < 60 chars (optional)
+  summary: A developer platform for building web, mobile, desktop, gaming, IoT, all your apps. Supported on Windows, Linux, and macOS. # < 160 chars (optional)
```
```suggestion
  summary: A developer platform for building web, mobile, desktop, gaming, IoT, and all your apps. Supported on Windows, Linux, and macOS. # < 160 chars (optional)
```

BillWagner replied:
I made a different edit here.

BillWagner committed:
proofread

Youssef1313 commented on index.yml:
```diff
@@ -0,0 +1,235 @@
+  title: Languages # < 60 chars (optional)
+  items:
+    # Card
+    - title: "C#: a modern object-oriented, and type-safe programming language in the C family."
```
```suggestion
    - title: "C#: a modern object-oriented, and type-safe programming language in the C family"
```

Youssef1313 commented on index.yml:
```diff
@@ -0,0 +1,235 @@
+      imageSrc: https://docs.microsoft.com/media/logos/logo_Csharp.svg
+      url: docs/csharp/index.md
+    # Card
+    - title: "Visual Basic:  an approachable language with a simple syntax for building type-safe, object-oriented apps."
```
```suggestion
    - title: "Visual Basic: an approachable language with a simple syntax for building type-safe, object-oriented apps"
```

Youssef1313 commented on index.yml:
```diff
@@ -0,0 +1,235 @@
+      imageSrc: https://docs.microsoft.com/media/logos/logo_VB.svg
+      url: docs/visual-basic/index.md
+    # Card
+    - title: "F#: a functional programming language for .NET, with object-oriented and imperative constructs."
```
```suggestion
    - title: "F#: a functional programming language for .NET, with object-oriented and imperative constructs"
```

Youssef1313 reviewed (changes requested):
In the previous commit, it seemed you was deleting the periods at end of titles. These weren't removed.

BillWagner committed:
another round of edits

The language text was too long.

Section summaries are complete sentences, and card summaries are phrases.

BillWagner committed:
make the VB and F# order consistent

BillWagner committed:
rework application types

BillWagner committed:
fix image links.

BillWagner committed:
move the index file to the docs folder

BillWagner commented:
@mairaw @mcleblanc 

This first mock up is ready. This PR would put this first version on a feature branch, not on master.

In this version, it moves the links from the current hub page to the new format, with duplicates removed.

Known tasks for the next update:

- [ ] Update images in the "Building your application" section. I've asked for the images used on http://dotnet/microsoft.com.
- [ ] Reconsider the content organization the *.NET: Free. Cross-platform.. Open source.* section.  (This includes several "get started" links.)
   - Is the .NET Guide / .NET Core Guide / .NET Framework Guide the right cards?
   - Or should it more task focused?

After those, we can start circulating for wider review.

mcleblanc commented on docs/index.yml:
```diff
@@ -0,0 +1,252 @@
+    - title: "Build your first web app. Learn how to build your first ASP.NET Core web app"
+      itemType: learn
+      url: https://dotnet.microsoft.com/learn/aspnet/hello-world-tutorial/intro
+
```
Why do we not link to the ASP.NET Core Get started tutorial (https://docs.microsoft.com/en-us/aspnet/core/getting-started/?view=aspnetcore-3.0), instead of the dotnet site Get started tutorial for the highlighted content section?

BillWagner replied:
@mcleblanc That's the current link on the [get-started pivot](https://docs.microsoft.com/en-us/dotnet/#pivot=docs&panel=getstarted). I'll change that in the next PR. 

I'd like the first commit on this branch to just convert to the new format. Then, we'll have a history of all the hub destination changes.

mcleblanc reviewed (approved):
Baseline conversion from existing hub looks good to me.
